dotCMS/src/main/java/com/dotmarketing/filters/VanityURLFilter.java

```diff
@@ -85,20 +86,10 @@ public void doFilter(final ServletRequest req, final ServletResponse res,
                     return;
                 }
 
-                /*
-                If the VanityURL has a query string we need to add it to the request in order to override
-                in the other filters.
-                 */
-                if (vanityUrlResult.getQueryString() != null) {
-                    request.setAttribute(CMS_FILTER_QUERY_STRING_OVERRIDE,
-                            vanityUrlResult.getQueryString());
-                }
+                filterChain.doFilter(new  VanityUrlRequestWrapper(request, vanityUrlResult) , response);
+                return;
+           }
 
-                /*
-                Set into the request the VanityURL we need to use to rewrite the current URI
-                 */
-                request.setAttribute(CMS_FILTER_URI_OVERRIDE, vanityUrlResult.getRewrite());
-            }
```

Comment:
really this chenge is not tested, I think this test is tested VanityURL maybe we can add a new query params test:

https://github.com/dotCMS/core/blob/master/dotCMS/src/integration-test/java/com/dotmarketing/filters/FiltersTest.java#L162